[vaidyanaath] iP

vaidyanaath opened the pull request:
# Duke
> "For every minute spent organizing, an hour is earned." -Anonymous

Say goodbye to cluttered thoughts and to-do lists. Duke is the ultimate solution for leading a more organized life.
Add tasks, manage them easily and keep track of upcoming deadlines or events.

It's
- text-based
- user-friendly
- ~~FAST~~ *BLAZINGLY* FAST ⚡ 

Follow these steps to get started:
1. download it from [here](https://github.com/vaidyanaath/ip/releases)
2. double-click it
3. add your tasks
4. sit back and manage your tasks with ease 😎

And it is absolutely **FREE!** 💸 

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [ ] Reminders (coming soon!)


If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
    start();
}
```

Commit:
Add Gradle support

vaidyanaath committed:
Add greet, echo, exit features

vaidyanaath committed:
Add print task-list and add task features

vaidyanaath committed:
Add mark as done feature

vaidyanaath committed:
Add todo, event, deadline features

vaidyanaath committed:
Add automated text UI testing

vaidyanaath committed:
Add error handling for user input

vaidyanaath committed:
Add error handling with exceptions

vaidyanaath committed:
Add delete task feature

vaidyanaath committed:
Add methods for representation of task states in data-file storage

vaidyanaath committed:
Add TaskFileHandler to read/write task-list in data-file storage

vaidyanaath committed:
Add data folder to gitignore

vaidyanaath committed:
Convert date strings to datetime objects

vaidyanaath committed:
Merge branch 'branch-Level-7'

vaidyanaath committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java

vaidyanaath committed:
Refactor user interaction methods into Ui class

vaidyanaath committed:
Rename TaskFileHandler to Storage class and refactor related methods

vaidyanaath committed:
Refactor task-list related methods into TaskList class, add Parser class

vaidyanaath committed:
Organize classes into packages

vaidyanaath committed:
Merge branch 'add-gradle-support' of https://github.com/vaidyanaath/ip into add-gradle-support

vaidyanaath committed:
Merge branch 'add-gradle-support'

vaidyanaath committed:
Add JUnit tests for Event, TaskList classes

vaidyanaath committed:
Add javadocs for task related classes

cydtseng commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,94 @@
+
+public class Storage {
+    private static final Path dataPath = Paths.get("data");
+    private static final Path taskListPath = Paths.get("data", "duke.txt");
```
Would be good to change constant names to adhere to UPPER_SNAKE_CASE.

vaidyanaath replied:
Thank you for taking the time to review my code and for bringing up this naming convention suggestion. I appreciate the suggestion and have made the necessary changes.

cydtseng reviewed (commented):
LGTM! :+1:
Suggested minor change.

vaidyanaath committed:
Modify code to follow coding standard

vaidyanaath committed:
Add getter method to access task description

vaidyanaath committed:
Add find feature

vaidyanaath committed:
Add javadocs for parser and storage classes

vaidyanaath committed:
Merge branch 'branch-A-JavaDoc'

vaidyanaath committed:
Merge branch 'branch-A-CodingStandard'

vaidyanaath committed:
Merge branch 'branch-Level-9'

vaidyanaath committed:
Fix variable naming convention in merged version

runoutofit commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,187 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    private final ArrayList<Task> TASK_LIST;
```
i like that you wrote the constant as UPPER_SNAKE_CASE

runoutofit commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,56 @@
+        }
+
+        // Multi-word commands
+        switch (command) {
```
i like that you implement switch case instead of multiple if-else statements which i see more often. i like how concise it makes your code look

runoutofit commented on src/main/java/Deadline.java:
```diff
@@ -1,14 +1,21 @@
 public class Deadline extends Task {
 
-    private String by;
+    private final String dueDate;
```
since dueDate is a constant, would upper snake case be more suitable to adhere to coding standard?

runoutofit commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,23 @@
+            + "|____/ \\__,_|_|\\_\\___|\n";
+
+    public void greet() {
+        String greeting = "____________________________________________________________\n"
```
would it be better to abstract out this line as a variable since you are using it often?

runoutofit reviewed (commented):
overall code is very good

vaidyanaath committed:
Add JavaFX dependencies

vaidyanaath committed:
Add empty UI using JavaFx

vaidyanaath committed:
Add basic echo feature to UI

vaidyanaath committed:
Add dialog boxes in UI with a stub response method

vaidyanaath committed:
Refactor Duke and TaskList.processCommand to return strings

vaidyanaath committed:
Remove text-ui classes

vaidyanaath committed:
Replace Duke stub response with working function

vaidyanaath committed:
Add changes to flip dialogBox UI

vaidyanaath committed:
Refactor UI using Java FXML

vaidyanaath committed:
Add checkstyle configuration with suppressions for code quality enforcement

vaidyanaath committed:
Merge branch 'branch-Level-10'

vaidyanaath committed:
Merge branch 'branch-A-CheckStyle'

vaidyanaath committed:
Minor build gradle changes

vaidyanaath committed:
Add assert statements

vaidyanaath committed:
Refactor loadTaskList() method

The loadTaskList() method is long and uses different layers of abstraction.

The code does not follow the SLAP principle. Refactoring the code with smaller methods also improves readability. The behaviour when there is an error with the data file is not made explicitly clear.

Let's,
* extract parseTaskList() and parseTask() methods from loadTaskList()
* extract clearing of data file into its own method

Doing this allows us to:
* understand better how the data file is processed
* change the behaviour of clearDataFile() method easily

vaidyanaath committed:
Rename variables in Task, Deadline, Event to follow coding standard

vaidyanaath committed:
Add descriptive variables for long boolean condition

vaidyanaath committed:
Add assert statements: processCommand() method, non null response

vaidyanaath committed:
Merge pull request #2 from vaidyanaath/branch-A-CodeQuality

Improve Code Quality

vaidyanaath committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/duke/storage/Storage.java

vaidyanaath committed:
Merge pull request #3 from vaidyanaath/branch-A-Assertions

Add assertions

vaidyanaath committed:
Set up Continuous Integration (CI)

Create gradle.yml

vaidyanaath committed:
Add javadocs for all public methods and checkstyle changes

vaidyanaath committed:
Edit tests to follow checkstyle

vaidyanaath committed:
Add sort by deadlines feature

The tasks are stored in the order in which they are created.

It is difficult for the user to quickly see which deadlines are due next.

Let's add a new command to sort the list of tasks chronologically based on the due date of the deadlines and preserve the order otherwise.

This change improves the usability of the chatbot application by allowing users to easily view and manage their tasks in order of their deadlines.

vaidyanaath committed:
Improve GUI: Add padding, title, icon, background color

vaidyanaath committed:
Fix bye command to close the GUI

vaidyanaath committed:
Merge branch 'branch-A-BetterGui'

vaidyanaath committed:
Add product Ui screenshot

vaidyanaath committed:
Update README.md
